python/lsst/ctrl/pool/pool.py

```diff
@@ -414,6 +415,60 @@ def log(self, source, msg, *args):
             self.out.write("\n")
 
 
+class ReductionThread(threading.Thread):
+    """Thread to do reduction of results
+
+    "A thread?", you say. "What about the python GIL?"
+    Well, because we 'sleep' when there's no immediate response from the
+    slaves, that gives the thread a chance to fire; and threads are easier
+    to manage (e.g., shared memory) than a process.
+    """
+    def __init__(self, reducer, initial=None, sleep=0.1):
+        threading.Thread.__init__(self, name="reducer")
+        self._queue = []  # Queue of stuff to be reduced
+        self._lock = threading.Lock()  # Lock for the queue
+        self._reducer = reducer  # Function that does the reducing
+        self._sleep = sleep  # Time to sleep when there's nothing to do (sec)
+        self._result = initial  # Final result
+        self._done = threading.Event()  # Signal that everything is done
+
+    def _doReduce(self):
+        """Do the actual work
+
+        We pull the data out of the queue and release the lock before
+        operating on it. This stops us from blocking the addition of
+        new data to the queue.
+        """
+        with self._lock:
+            queue = self._queue
+            self._queue = []
+        for data in queue:
+            self._result = self._reducer(self._result, data) if self._result is not None else data
```

Comment:
Serious question: are we shooting for < 80 characters now? or waiting until we've run autopep8?